### PR TITLE
Remember the last saved or last opened file location

### PR DESCRIPTION
**Pull request by Auriga2**
### Description of the Change
The last directory accessed for opening or saving a graph is now saved and remembered between sessions as well as within a session. Changed the preference to `Remember Open/Save Location` in `Setup`->`Options`->`CONSTELLATION`->`Save Location`, which is enabled by default. When this is not set, both the open and save locations will be the user's home directory.

### Alternate Designs
N/A

### Why Should This Be In Core?
Better UX

### Benefits
Saves time.

### Possible Drawbacks
N/A

### Verification Process

1. First check the preference to remember the last saved location is turned on. Go to Options > Constellation > `Open/Save Location`. Ensure `Remember Open/Save location` is ticked
2. Create and save a new graph. Take note of the saved location.
3. Open up another new graph, draw a node and click to save (Using Save, Save as or Ctrl+s). Verify the previous location is opened.
4. Try to open a previously saved STAR file using File-> open on or Ctrl+o, and verify the same location is opened on the Open File window.
5. Try to save a file now, and verify the previously opened location is opened up in the Save file window.
6. Close constellation, noting the last opened or saved location. Run it again and repeat again. Verify both Open and Save windows open up with this location.
7. Open a file in the Welcome page and repeat the same above.
8. Repeat again, changing the opened and saved location in between.
9. Repeat again with the preference unticked and verify the user's home directory is opened up.

### Applicable Issues
https://github.com/constellation-app/constellation/issues/1098

### CODE REVIEW

**Review by arcturus2 — APPROVED** [review]
Thank you @Auriga2 it works well.

File: CorePreferences/src/au/gov/asd/tac/constellation/preferences/ApplicationPreferenceKeys.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ private static String getDefaultUserDir() {
     public static final String JUPYTER_NOTEBOOK_DIR_DEFAULT = new File(System.getProperty("user.dir")).getPath();
 
     /**
-     * Remember Save As location.
+     * Remember Open and Save location.
      */
     public static final String REMEMBER_SAVE_LOCATION = "rememberSaveLocation";
     public static final boolean REMEMBER_SAVE_LOCATION_DEFAULT = true;
```
Comment (antares1470) [inline CorePreferences/src/au/gov/asd/tac/constellation/preferences/ApplicationPreferenceKeys.java:113]: I feel like these should be renamed to `REMEMBER_OPEN_SAVE_LOCATION` and `REMEMBER_OPEN_SAVE_LOCATION_DEFAULT` to reflect that it is for both open and save now (not just save)

**Review by antares1470 — CHANGES_REQUESTED** [review]
Looks good in testing

File: CorePreferences/src/au/gov/asd/tac/constellation/preferences/ApplicationPreferenceKeys.java (hunk 1 above)
Comment (Auriga2) [reply CorePreferences/src/au/gov/asd/tac/constellation/preferences/ApplicationPreferenceKeys.java:113]: @antares1470 I changed these (manually), but it threw an error, so I reverted them back. Is there a different way to rename?

Comment (antares1470) [reply CorePreferences/src/au/gov/asd/tac/constellation/preferences/ApplicationPreferenceKeys.java:113]: @Auriga2 I tried this locally on my machine and I didn't seem to have any such troubles. Did you refactor the variable name or were you manually changing it at all occurrences?

Comment (Auriga2) [reply CorePreferences/src/au/gov/asd/tac/constellation/preferences/ApplicationPreferenceKeys.java:113]: I can't refactor, get this error:
image

So I manually edited. Could be a manual error that broke it. Can you see the prefs in the `Setup->Options->CONSTELLATION` tab after refactoring?

Comment (Auriga2) [reply CorePreferences/src/au/gov/asd/tac/constellation/preferences/ApplicationPreferenceKeys.java:113]: Oh I can refactor, had to open the module before refactoring, as suggested by the message above. I'll see if it works now.

Comment (Auriga2) [reply CorePreferences/src/au/gov/asd/tac/constellation/preferences/ApplicationPreferenceKeys.java]: Refactoring worked. I changed other occurrences as well.

**Review by antares1470 — APPROVED** [review]
Once merge conflicts are resolved this should be good to go

**Comment by Auriga2** [comment]
Looks like I can fix one code smell. Shall I push the branch again?
image

I guess the bug is not an issue here as we handle the null file. What do you think?
image

**Comment by antares1470** [comment]
Oh my bad, yeah push again @Auriga2

**Comment by Auriga2** [comment]
Pushed the branch again, although I can't see it linked here.

**Comment by antares1470** [comment]
> 
> 
> Pushed the branch again, although I can't see it linked here.

The branch was deleted when I merged the PR so you may need to create a new PR

**Pull request merged**